src/commands/blocking_commander.h

```diff
@@ -68,6 +69,15 @@ class BlockingCommander : public Commander,
   }
 
   void OnWrite(bufferevent *bev) {
+    {
+      // The blocking command should not be executed when the server is in exclusive state,
+      // because it might have the data race when the server is in transaction mode and run
+      // the callback here might cause the current execution also in transaction mode.
+      //
+      // For more context, please refer to: https://github.com/apache/kvrocks/issues/2900
+      auto concurrency = conn_->GetServer()->WorkConcurrencyGuard();
```

Comment:
It will be destructed immediately if we put it here, we can put it before `auto guard = GetLocks();`.